pkg/operator/generic_client.go

```diff
@@ -87,3 +95,122 @@ func (p *genericClient) UpdateOperatorStatus(ctx context.Context, resourceVersio
 
 	return &ret.Status.OperatorStatus, nil
 }
+
+func (p *genericClient) ApplyOperatorSpec(ctx context.Context, fieldManager string, desiredConfiguration *v1.OperatorSpecApplyConfiguration) (err error) {
+	if desiredConfiguration == nil {
+		return fmt.Errorf("applyConfiguration must have a value")
+	}
+
+	desiredSpec := &v1.OpenShiftControllerManagerSpecApplyConfiguration{
+		OperatorSpecApplyConfiguration: *desiredConfiguration,
+	}
+	desired := v1.OpenShiftControllerManager("cluster")
+	desired.WithSpec(desiredSpec)
+
+	instance, err := p.client.OpenShiftControllerManagers().Get(ctx, "cluster", metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+	// do nothing and proceed with the apply
+	case err != nil:
+		return fmt.Errorf("unable to get operator configuration: %w", err)
+	default:
+		original, err := v1.ExtractOpenShiftControllerManager(instance, fieldManager)
+		if err != nil {
+			return fmt.Errorf("unable to extract operator configuration: %w", err)
+		}
+		if equality.Semantic.DeepEqual(original, desired) {
+			return nil
+		}
+	}
+
+	_, err = p.client.OpenShiftControllerManagers().Apply(ctx, desired, metav1.ApplyOptions{
+		Force:        true,
+		FieldManager: fieldManager,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to Apply for operator using fieldManager %q: %w", fieldManager, err)
+	}
+
+	return nil
+}
+
+func (p *genericClient) ApplyOperatorStatus(ctx context.Context, fieldManager string, desiredConfiguration *v1.OperatorStatusApplyConfiguration) (err error) {
+	if desiredConfiguration == nil {
+		return fmt.Errorf("applyConfiguration must have a value")
+	}
+
+	instance, err := p.client.OpenShiftControllerManagers().Get(ctx, "cluster", metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+		// do nothing and proceed with the apply
+		v1helpers.SetApplyConditionsLastTransitionTime(p.clock, &desiredConfiguration.Conditions, nil)
+		desiredStatus := &v1.OpenShiftControllerManagerStatusApplyConfiguration{
```

Comment:
This and the default block have some code duplication. Worth de-duplicating after the switch.

Reply:
Deduplicated some code portions. Thanks.

Reply:
@ingvagabund CI passed. Could you PTAL one more round?.